### PR TITLE
task: add support for guest checkout buyer

### PR DESCRIPTION
**Pull request by luca-gr4vy**
**Description:** Adds support for guest checkout by introducing a new `buyer` option.

**Ticket:** https://gr4vy.atlassian.net/browse/TA-8175

**Example usage (manually tested):**

```
buyer: Gr4vyBuyer(
    displayName: "Test buyer",
    billingDetails: Gr4vyBillingDetails(
        firstName: "Ivy",
        lastName: "Leaf",
        address: Gr4vyAddress(
            city: "London"
        ),
        taxId: Gr4vyTaxId(
            value: "123456",
            kind: "gb.vat"
        )
    )
)
```

### CODE REVIEW

File: gr4vy-iOS/Models/Gr4vyBuyer.swift [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        shippingDetails: Gr4vyBillingShippingDetails? = nil
+    ) {
+        self.displayName = displayName
+        self.externalIdentifier = externalIdentifier
+        self.billingDetails = billingDetails
+        self.shippingDetails = shippingDetails
+    }
+}
+
+public struct Gr4vyBillingShippingDetails: Codable {
```
Comment (luca-gr4vy) [inline gr4vy-iOS/Models/Gr4vyBuyer.swift:27]: Combining the two `billingDetails` and `shippingDetails` with the same properties

Comment (victormferrara) [reply gr4vy-iOS/Models/Gr4vyBuyer.swift:27]: As in https://github.com/gr4vy/gr4vy-android/pull/14, maybe these should be different as the shipping details don't have the `taxId`?

Comment (luca-gr4vy) [reply gr4vy-iOS/Models/Gr4vyBuyer.swift]: updated

File: gr4vy-iOSTests/gr4vy_iOSTests.swift [hunk 2]
```diff
@@ -1128,6 +1128,49 @@ class gr4vy_iOSTests: XCTestCase {
...
+        XCTAssertEqual("window.postMessage({ \"channel\": 123, \"type\": \"updateOptions\", \"data\": {\"amount\":100,\"apiHost\":\"api.ID123.gr4vy.app\",\"apiUrl\":\"https:\\/\\/api.ID123.gr4vy.app\",\"buyer\":{\"billingDetails\":{\"address\":{\"city\":\"city\",\"country\":\"country\",\"houseNumberOrName\":\"houseNumberOrName\",\"line1\":\"line1\",\"line2\":\"line2\",\"organization\":\"organization\",\"postalCode\":\"postalCode\",\"state\":\"state\",\"stateCode\":\"stateCode\"}}},\"country\":\"GB\",\"currency\":\"GBP\",\"supportedApplePayVersion\":0,\"token\":\"TOKEN123\"}})", sut)
+
+        setup.buyer = Gr4vyBuyer(billingDetails: Gr4vyBillingDetails(taxId: Gr4vyTaxId(value: "value", kind: "kind")))
+
+        sut = Gr4vyUtility.generateUpdateOptions(from: setup)
+        XCTAssertEqual("window.postMessage({ \"channel\": 123, \"type\": \"updateOptions\", \"data\": {\"amount\":100,\"apiHost\":\"api.ID123.gr4vy.app\",\"apiUrl\":\"https:\\/\\/api.ID123.gr4vy.app\",\"buyer\":{\"billingDetails\":{\"taxId\":{\"kind\":\"kind\",\"value\":\"value\"}}},\"country\":\"GB\",\"currency\":\"GBP\",\"supportedApplePayVersion\":0,\"token\":\"TOKEN123\"}})", sut)
+    }
```
Comment (victormferrara) [inline gr4vy-iOSTests/gr4vy_iOSTests.swift:1173]: Maybe add one case using shipping details?

Comment (luca-gr4vy) [reply gr4vy-iOSTests/gr4vy_iOSTests.swift:1178]: added, thanks

**Review by victormferrara — APPROVED** [review]

**Pull request merged**